src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclint/Checker.java

```diff
@@ -690,7 +690,9 @@ public Void visitAttribute(AttributeTree tree, Void ignore) {
             }
             // for now, doclint allows all attribute names beginning with "on" as event handler names,
             // without checking the validity or applicability of the name
-            if (!name.toString().startsWith("on")) {
+            // custom "data-*" attributes are also accepted
+            var attrName=name.toString();
+            if (!attrName.startsWith("on") && !attrName.matches("data-[a-z]+(-[a-z]+)*")){
```

Comment:
I understand that `startsWith("on")` precedes this PR. However, it's more permissive than it should be: in reality, there are only 70 global attributes that start with "on". On the other hand, `attrName.matches("data-[a-z]+(-[a-z]+)*")` is less permissive. For example, if I read the XML spec correctly, such an attribute can contain `_` or start with `data-.`. But maybe it's good enough.

Reply:
I like being slightly restrictive and safe, maybe we can even use a set of event attributes here? If 70 elements isn't too much.

Reply:
As @jonathan-gibbons likes to point out, javadoc is not an HTML validation tool. So I think, it's okay to leave the code simple. Maybe this would be even simpler?

    data-[a-z][-a-z0-9]*

Reply:
I admit to being "lazy" when I wrote `startsWith("on")`. In general, I don't think folk should be writing many if any event handlers in doc comments, and it was easier to give forward compatibility with a weak test for `"on"`. Likewise, there should not be many if any uses of `data-*` attributes, and it is easy enough to give a simple regex. It's more important to not reject valid uses of event attributes and custom data attributes than it is to detect all invalid cases. The right place to detect all invalid cases is downstream in recognized validation tools, like `tidy` which catch all occurrences of bad stuff, wherever they originate (doc comment, command-line options, the `javadoc` tool, post-processors, etc).  DocLint is primarily an "early warning system" for common errors.

Reply:
I agree with you, Jon. So, we can simply use `startsWith("data-")`, which would be good enough and most permissive (i.e. no false negatives).

Reply:
Sure, I will go with that.